pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -260,7 +260,9 @@ func (r *CiliumNetworkPolicy) Parse() (api.Rules, error) {
 	if r.Spec != nil {
 		if err := r.Spec.Sanitize(); err != nil {
 			return nil, fmt.Errorf("Invalid CiliumNetworkPolicy spec: %s", err)
-
+		}
+		if namespace != "" && r.Spec.NodeSelector.LabelSelector != nil {
+			return nil, fmt.Errorf("NodeSelector is only valid in CiliumClusterwideNetworkPolicy")
```

Comment:
Nit: simple error strings can be created using `errors.New` instead of `fmt.Errorf`. This will help in reducing the usage of `fmt` and eventually the need to import it, but it seems like we already use `fmt.Errorf` in this file so this is a moot point. Feel free to ignore

Reply:
heheh, interesting. I think in other reviews we have encouraged use of `fmt.Errorf` so we can get rid of `import errors` ;)

Is there some "official"(ish) Go community guidance on this at all?

Reply:
:hot_pepper: interesting! I wasn't aware that there was an effort to move away from `errors`. 

In all seriousness, I haven't seen an official stance from the community, but it is known throughout the Golang ecosystem, that `fmt` is a heavy package that bloats binaries. 

Here's a quick test that I did:

```
❯ cat main-fmt.go
package main

import "fmt"

func main() {
        _ = fmt.Errorf("error")
}

/tmp/errors
❯ cat main-errors.go
package main

import "errors"

func main() {
        _ = errors.New("error")
}

/tmp/errors
❯ ll
total 3.2M
-rwxr-xr-x 1 chris chris 1.3M May 21 13:55 errors
-rwxr-xr-x 1 chris chris 2.0M May 21 13:55 fmt
-rw-r--r-- 1 chris chris   72 May 21 13:54 main-errors.go
-rw-r--r-- 1 chris chris   69 May 21 13:54 main-fmt.go
```

You can see the `fmt` binary is significantly larger than the `errors` binary. However, one caveat for this test is that this test is very simple and naive. In reality, the Cilium binaries may just have to pay the cost of importing the `fmt` package once. I don't know if that's true or not, but I'd love to know. If we pay the cost once, then great we can use it liberally. But if we don't then it'd be good to slowly start moving away from `fmt.Errorf` where we can.

Reply:
I'd rather not implement that change as part of this PR. I think someone needs to look into it and make a tree-wide change, maybe with something to enforce. Otherwise, we're just going to go back and forth between `fmt` and `errors`...